google-cloud-firestore/lib/google/cloud/firestore/collection_reference_list.rb

```diff
@@ -0,0 +1,137 @@
+        #
+        def next
+          return nil unless next?
+          ensure_service!
+          grpc = @client.service.list_collections @parent, token: token, max: @max
+          self.class.from_grpc grpc, @client, @parent, max: @max
+        end
+
+        ##
+        # Retrieves remaining results by repeatedly invoking {#next} until {#next?} returns `false`. Calls the given
+        # block once for each result, which is passed as the argument to the block.
+        #
+        # An Enumerator is returned if no block is given.
+        #
+        # This method will make repeated API calls until all remaining results are retrieved. (Unlike `#each`, for
+        # example, which merely iterates over the results returned by a single API call.) Use with caution.
+        #
+        # @param [Integer] request_limit The upper limit of API requests to make to load all collection references.
+        #   Default is no limit.
+        # @yield [collection_reference] The block for accessing each collection_reference.
+        # @yieldparam [CollectionReference] collection_reference The collection reference object.
+        #
+        # @return [Enumerator]
+        #
+        # @example Iterating each collection reference by passing a block:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   firestore.cols do |collection_reference|
+        #     puts collection_reference.collection_id
+        #   end
+        #
+        # @example Using the enumerator by not passing a block:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   all_collection_ids = firestore.cols.map do |collection_reference|
+        #     collection_reference.collection_id
+        #   end
+        #
+        def all request_limit: nil
+          request_limit = request_limit.to_i if request_limit
+          unless block_given?
+            return enum_for :all, request_limit: request_limit
+          end
+          results = self
+          loop do
+            results.each { |r| yield r }
+            if request_limit
+              request_limit -= 1
+              break if request_limit < 0
+            end
+            break unless results.next?
+            results = results.next
+          end
+        end
+
+        ##
+        # @private New CollectionReference::List from a `Google::Cloud::Firestore::V1::ListCollectionIdsResponse`
+        # object.
+        def self.from_grpc grpc, client, parent, max: nil
+          raise ArgumentError, "parent is required" unless parent
+          cols = CollectionReferenceList.new(Array(grpc.collection_ids).map do |collection_id|
+            CollectionReference.from_path "#{parent}/#{collection_id}", client
+          end)
+          token = grpc.next_page_token
+          token = nil if token == "".freeze
```

Comment:
Just curious why the string is frozen.

Reply:
I suspect it was done to appease Rubocop back in the day. Removing.

Reply:
Done, along with a few similar usages.